### PR TITLE
Changes to enable Nitro enclaves

### PR DESCRIPTION
**Pull request by dreemkiller**
Added a new docker target nitro. 
Certain things needed to be done inside the make file (like building the AWS nitro-cli tools) instead of the docker file as I would have liked.

### CODE REVIEW

File: Dockerfile
Comment (ericvh) [inline Dockerfile]: nice

File: Dockerfile
Comment (ericvh) [inline Dockerfile]: I've got a slightly more secure variation of this I use:
        if [ "$USER" != "root" ] ; then useradd -u $UID -m -p `openssl rand -base64 32` -s /bin/bash $USER ; fi; \

File: Dockerfile
Comment (ericvh) [inline Dockerfile]: Am I reading this right?  Are you always installing the nitro stuff in base?  Should we isolate it?

File: Makefile
Comment (ericvh) [inline Makefile]: does nitro really need the X11 stuff?

File: Dockerfile
Comment (dreemkiller) [reply Dockerfile]: That was a test. You passed.

File: Dockerfile
Comment (dreemkiller) [reply Dockerfile]: probably so. I'm going to do a rebase on top of your changes, and I'll move it.

File: Makefile
Comment (dreemkiller) [reply Makefile]: It does not.

**Comment by ericvh** [comment]
rebase is going to be a bit odd, since what i did was split the target
specific docker files to their own subdirs -- so basically I 'accepted all'
in the main Dockerfile and then just moved the new stuff to a subdir.

**Comment by dreemkiller** [comment]
Yeah, I was using the term "rebase" loosely. More like "rewrite everything on top of the latest version".

**Comment by dreemkiller** [comment]
Yeah, I was using the term "rebase" loosely. More like "rewrite everything on top of the latest version".

That's done now.

File: Makefile
Comment (nspin) [inline Makefile]: Pinning a revision or version branch could make this part more reproducible. Not necessary of course but we have talked about striving for reproducibility.

(I'm talking about reproducibility in the informal sense, not the bit-for-bit sort we've also discussed)

Comment (dreemkiller) [reply Makefile]: That's a good idea. Will do.

Comment (dreemkiller) [reply Makefile]: I've done this now. Take a look

**Comment by dreemkiller** [comment]
@nspin Is there anything holding this up now?

**Comment by dreemkiller** [comment]
Can we get this approved? @ShaleXIONG ? It's been sitting here a while.

**Comment by dreemkiller** [comment]
@DominicPM Do you mind reviewing this? I cant' get any action on it.

**Comment by dominic-mulligan-arm** [comment]
Sure, I'll take a look.

File: Makefile
Comment (dominic-mulligan-arm) [inline Makefile:9]: Is this particular revision important?

**Comment by dominic-mulligan-arm** [comment]
Skimmed through manually and only have one comment.  I'll checkout locally and try to build, too.

**Review by ShaleXIONG — APPROVED** [review]

**Comment by dominic-mulligan-arm** [comment]
@ShaleXIONG did you check that this builds correctly?  I seem to remember you did.  If so, I don't need to do it and this can be merged.

**Comment by ShaleXIONG** [comment]
> @ShaleXIONG did you check that this builds correctly? I seem to remember you did. If so, I don't need to do it and this can be merged.

Yes, I checked it a long time ago, forgot to approve.

**Review by nspin — APPROVED** [review]

File: Makefile
Comment (dominic-mulligan-arm) [inline Makefile:54]: A stray mention of Tabasco in `TABASCO_IP_ADDRESS`.

Comment (dominic-mulligan-arm) [reply Makefile:54]: Is that even needed any more?

File: Makefile
Comment (dreemkiller) [reply Makefile:9]: Not particularly. But I added the specific version at @nspin 's suggestion to make builds repeatable (which is a Good Idea).

Comment (dominic-mulligan-arm) [reply Makefile:9]: OK, that makes sense.

File: Makefile
Comment (dreemkiller) [reply Makefile:54]: Yes, still needed until #180 is merged. The name is anachronistic, but exists in current code on main until the #180 merge.

Comment (dominic-mulligan-arm) [reply Makefile:54]: OK, cool.

**Review by dominic-mulligan-arm — APPROVED** [review]

**Comment by dominic-mulligan-arm** [comment]
+1+1+1 = +1 reached.  merging.

**Pull request merged**
